numpy/lib/function_base.py

```diff
@@ -2669,6 +2689,23 @@ class vectorize(object):
            <http://docs.scipy.org/doc/numpy/reference/c-api.generalized-ufuncs.html>`_.
     """
 
+    # __new__ is overriden as a clean way to provide decorator-with-keywords
+    # syntax while allowing 'vectorize' to remain as the type of vectorized
+    # functions.
+    def __new__(cls, pyfunc=None, *args, **kwargs):
+
+        def vectorize_decorator(pyfunc):
+            """ ``vectorize`` with presupplied keyword arguments. """
+            # prevent an edge case where None would produce another decorator
+            if not hasattr(pyfunc, '__call__'):
+                raise TypeError('the wrapped object must be callable')
+            return cls(pyfunc, *args, **kwargs)
+
+        if pyfunc is None:
+            return vectorize_decorator
+        else:
+            return super(vectorize, cls).__new__(cls)#, *args, **kwargs)
```

Comment:
oops. (tsk, I commented this out with the intent to make a test fail. None do...)